heroku-deploy/src/main/java/com/heroku/sdk/deploy/lib/deploymemt/Deployer.java

```diff
@@ -0,0 +1,118 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+public final class Deployer {
+
+    public static boolean deploy(String apiKey, String clientName, String clientVersion, DeploymentDescriptor deploymentDescriptor, OutputAdapter outputAdapter) throws IOException, InterruptedException {
+        HerokuAPI herokuApi = new HerokuAPI(apiKey);
+        HerokuDeployApi herokuDeployApi = new HerokuDeployApi(clientName, clientVersion, apiKey);
+
+        Source source = herokuApi.createSource();
+        Source.Blob sourceBlob = source.getSource_blob();
+
+        if (!deploymentDescriptor.getConfigVars().isEmpty()) {
+            outputAdapter.logInfo("-----> Setting config vars...");
+            try {
+                herokuApi.updateConfig(deploymentDescriptor.getAppName(), deploymentDescriptor.getConfigVars());
+            } catch(Throwable t) {
+                outputAdapter.logError("Could not set config vars: " + t.getMessage());
+                return false;
+            }
+        }
+
+        outputAdapter.logInfo("-----> Uploading build...");
+        uploadSourceBlob(deploymentDescriptor.getSourceBlobPath(), URI.create(sourceBlob.getPut_url()), outputAdapter::logUploadProgress);
+        outputAdapter.logInfo("       - success");
+
+        List<String> buildpacks = deploymentDescriptor.getBuildpacks();
+        if (buildpacks.isEmpty()) {
+            buildpacks = Collections.singletonList("heroku/jvm");
+        }
+
+        outputAdapter.logInfo("-----> Deploying...");
+        BuildInfo buildInfo;
+        try {
+            buildInfo = herokuDeployApi.createBuild(
+                    deploymentDescriptor.getAppName(),
+                    URI.create(sourceBlob.getGet_url()),
+                    deploymentDescriptor.getVersion(),
+                    buildpacks);
+        } catch (AppNotFoundException e) {
+            outputAdapter.logError("Could not find application! Make sure you configured your application name correctly.");
+            return false;
+        }  catch (InsufficientAppPermissionsException e) {
+            outputAdapter.logError("Insufficient permissions to deploy to application! Make sure you configured your application name correctly.");
+            return false;
+        } catch (HerokuDeployApiException e) {
+            outputAdapter.logError("Unknown error while deploying: " + e.getMessage(), e);
+            return false;
+        }
+
+        try {
+            herokuDeployApi
+                    .followBuildOutputStream(URI.create(buildInfo.outputStreamUrl))
+                    .map(line -> "remote: " + line)
+                    .forEachOrdered(outputAdapter::logInfo);
+        } catch (SSLHandshakeException e) {
+            outputAdapter.logWarn("Could not get remote output. You might run an older Java version without Let's Encrypt support. Your build will continue to run, stand by.");
```

Comment:
might want to rephrase this. it almost sounds like you're recommending they use an older version. How about "You could be running an older Java version..."

Reply:
Thanks! 👍